crates/polars-plan/src/logical_plan/builder.rs

```diff
@@ -239,21 +237,61 @@ impl LogicalPlanBuilder {
         cache: bool,
         row_index: Option<RowIndex>,
         rechunk: bool,
+        #[cfg(feature = "cloud")] cloud_options: Option<CloudOptions>,
     ) -> PolarsResult<Self> {
-        use polars_io::SerReader as _;
+        use polars_io::is_cloud_url;
 
         let path = path.into();
-        let file = polars_utils::open_file(&path)?;
-        let mut reader = IpcReader::new(file);
 
-        let reader_schema = reader.schema()?;
-        let mut schema: Schema = (&reader_schema).into();
-        if let Some(rc) = &row_index {
-            let _ = schema.insert_at_index(0, rc.name.as_str().into(), IDX_DTYPE);
-        }
+        let (schema, reader_schema, num_rows, metadata) = if is_cloud_url(&path) {
+            #[cfg(not(feature = "cloud"))]
+            panic!(
+                "One or more of the cloud storage features ('aws', 'gcp', ...) must be enabled."
+            );
+
+            #[cfg(feature = "cloud")]
+            {
+                let uri = path.to_string_lossy();
+                get_runtime().block_on(async {
+                    let reader =
+                        polars_io::ipc::IpcReaderAsync::from_uri(&uri, cloud_options.as_ref())
+                            .await?;
+                    let metadata = reader.metadata().await?;
+                    let reader_schema = Arc::clone(&metadata.schema);
+
+                    let schema = prepare_schema((&reader_schema).into(), row_index.as_ref());
+                    PolarsResult::Ok((schema, reader_schema, None, metadata))
+                })?
+            }
+        } else {
+            // NOTE: We do not really need IpcReader and all of it's memoization
+            // to read the metadata...
+            let mut file = polars_utils::open_file(&path)?;
+
+            let metadata = arrow::io::ipc::read::read_file_metadata(&mut file)?;
+            let reader_schema = Arc::clone(&metadata.schema);
+            let mut schema: Schema = (&reader_schema).into();
+            if let Some(rc) = &row_index {
+                let _ = schema.insert_at_index(0, rc.name.as_str().into(), IDX_DTYPE);
+            }
```

Comment:
This looks like what `prepare_schema` is doing, we can probably replace this with a call to it.